[DNM] TG Slaughter Demon port

Aurorablade opened the pull request:
Ports TG Slaughter Demon by @KorPhaeron https://github.com/tgstation/-tg-station/pull/9973 (is it bad i have been stalking this for days to port it?) All credits to him for the concept and more credits i shoved in the file itself.


>New side event/antagonist. Simple animal that can use pools of blood as
>portals to move around the station. Spooky scary.

>-CtrlClick on an adjacent blood pool (or turf with blood on it) to
>"enter" (start an infinite duration jaunt
>-CtrlClick on a blood pool while jaunted to exit at that pool
>-If you're pulling a mob while you enter a pool, the mob gets pulled in
>with you, and you devour them, healing yourself
>-Night vision

>Adds an event (admin only) that spawns one.

>Adds wizard item for summoning it. Has objective to kill the wizard (and everyone else).


What I have done:
-Ported
It works(admin and wizard spawn only)
-Added a bit more blood
-Changed Verbage on some objectives
-Did a reverse animation based on the jaunt animation for the daemon so it rises out of the blood.
-Did the whole 'would you like to play this?' question for getting the antag.
-Added in a Whisper abilty so you can creepy whisper people...

What i would like to do:
-Find places to add more blood(i need suggestions)

-Tweak on suggestion to make it fit the server and not as murderboneish(though it IS A hell demon)

-Maybe tweak the wizard bit, as from what i know on paradise once the wizard is kill round is kill. (Creatively the wizard could trick a crew member into summoning it or thier apprentiace. "Here break this for me, student..")

-MORE BLOOD
-Cooldown on entering/exiting blood?
-I would love when the demon is done eating to eject the players brain and a random organ (in hopes of borg/cloneing) but don't know how to do it.
-go from ctrl click to alt click?
-Fix any glareing errors in this 
-MORE BLOOD ALL THE BLOOD


Silly bonus:
Adds in the Honk Render and Crab render (I saw the honk render on tg...and for some reason wanted one for crabs..)
I am sure these items will be used in a responisble manner by the admins...*Adminbus honks in the distance.*

And now a visual representation of this PR:
![space station 13 2015-06-16 154208](https://cloud.githubusercontent.com/assets/5515461/8192948/85efff00-143f-11e5-9e1d-75e4119925d5.png)

Robertha89 commented:
:+1: x :100:

KorPhaeron commented:
Hurray I'm famous

Aurorablade commented:
yeah alot of stuff is not working like i want but hopefully someone can push me in the right direction. or use the clow method and knock me over and repeatedly pull me over a banana peel.

Robertha89 commented:
for blood, check the vampiric here https://github.com/ParadiseSS13/Paradise/blob/6f69d256235c95048cc735385fe437908f66baf3/code/modules/research/xenoarchaeology/finds/finds_special.dm

Spacemanspark commented:
Ahahaha I was just looking at that PR.
:+1:

Fox-McCloud commented:
I'm not opposed, at all to this mini-antag being ported--as a matter of fact, I encourage it. That said, as this is your second PR and you seem to need a lot of help with this, I do encourage you to perhaps tackle a smaller project first. If you really think you're up for this, that's fine, but if you think it may be too much for you, then you may want to work on a smaller port or own self-created project.

Again, not downing you at all and I'm *NOT* opposed to this making it into the codebase--it's just that this is a pretty big project fora second PR and it may be a bit overwhelming for you.

Aurorablade commented:
@Fox-McCloud I understand what you are saying completely. I am pretty used to coding and i have this bad habit of jumping in head first with a new language/codebase and pushing all the shiny buttons until i figure out what they do. also just trying to keep this to acceptable standards of omg don't muderbone me bro while still be creepy. That said i was looking at your Port of revernents its giving me a bit of idea/example to go off of as this code seems to use that as a basis so thoughts so far are:
This was intended as an admin spawn only event, should i just leave it to the event panel as a weight of 0 instead of trying to figure out the one click antag?
On the spawned version rather then the vial version, should i revert BACK to not asking a ghost if they want to be the Daemon and just going 'DING YOUR A DAEMON NOW GHOSTY POTTER'?
CTRL+click or ALT+click?

I think i am close to understanding the droplimb/organ dropping stuff i just need to stare at some code bits a bit more...

And does anyone have any suggestions on how to add MORE BLOOD? the vamp statuette gives me an idea but..

Robertha89 commented:
gib spawner on entering and leaving blood pools, it would create lots of localised blood

Aurorablade commented:
Updated commit and added a visual representation fo the PR....

Fox-McCloud commented on code/modules/events/event_container.dm:
```diff
@@ -169,6 +169,7 @@ var/global/list/severity_to_string = list(EVENT_LEVEL_MUNDANE = "Mundane", EVENT
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Bluespace Anomaly",		/datum/event/anomaly/anomaly_bluespace,	50,		list(ASSIGNMENT_ENGINEER = 25)),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Flux Anomaly",				/datum/event/anomaly/anomaly_flux,		50,		list(ASSIGNMENT_ENGINEER = 50)),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Gravitational Anomaly",	/datum/event/anomaly/anomaly_grav,		200),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Slaughter Demon",			/datum/event/spawn_slaughter,		0),
```
No point to ading this if it's an admin only event; admins with necessary permissions can trigger events manually.

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+	health = 250
+	environment_smash = 1
+	universal_understand = 1
+	melee_damage_lower = 15//Paradise port:This was also 30, but i moved it down.
```
I'd leave this at 30.

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+	return 1
+
+
+///obj/effect/dummy/slaughter/singularity_act(blah)//this errored on me on compile - Aurora
```
your branch is outdated; you need to update it--that's why this pitched a fit.

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+		new /obj/effect/gibspawner/human(get_turf(src))
+		playsound(get_turf(src), 'sound/misc/exit_blood.ogg', 100, 1, -1)
+		qdel(src.holder)
+		del(animation)
```
qdel(animation)

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+			if(!(src.eating))
+				new /obj/effect/gibspawner/human(get_turf(src))///Somewhere a janitor weeps..
+				sleep(6)///wait till the anim finishs before we delet it..
+				del(animation)
```
qdel(animation)

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+				src.visible_message("<span class='warning'>Loud eating sounds come from the blood...</B>")
+				src.eating = TRUE
+				sleep(6)///wait till the anim finishs before we delet it..
+				del(animation)
```
qdel(animation)

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+				sleep(30)
+				var/mob/living/carbon/human/nomNom = src.kidnapped
+				var/obj/item/organ/brain/brainObj = nomNom.internal_organs_by_name["brain"]
+				if(brainObj && istype(brainObj))
```
Bitty iff on this not being a permadeath.

Aurorablade replied:
@Fox-McCloud Well i wasn't sure if perma death would be alright or not. I can change it to try to throw out a random organ.

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+			animation.dir = src.dir
+			src.ExtinguishMob()
+			if(src.buckled)
+				src.buckled = null
```
src.buckled.unbuckle() instead

Fox-McCloud commented on code/game/gamemodes/wizard/spellbook.dm:
```diff
@@ -386,6 +392,11 @@
+	if(istype(O, /obj/item/weapon/antag_spawner/slaughter_demon))
+		user << "<span class='notice'>On second thought, maybe summoning a demon is a bad idea. You refund your points.</span>"
+		src.uses++
+		del(O)
```
qdel(O)

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+/mob/living/simple_animal/slaughter/New()
+	..()
+	spawn(5)
+	AddSpell(new /obj/effect/proc_holder/spell/wizard/targeted/slaughter_whisper)
```
ehhh...not sure if this is the best way to handle it or if it should be added to the mind.

Aurorablade replied:
I will try adding it in to mind again.

Fox-McCloud commented on code/modules/mob/living/simple_animal/slaughter/slaughter.dm:
```diff
@@ -0,0 +1,274 @@
+				if(istype(src.pulling, /mob/living/))
+					var/mob/living/victim = src.pulling
+					var/mob/living/carbon/human/victimType = src.pulling
+					if (victimType.species.flags & IS_SYNTHETIC)//IPCs are too crunchy....I am also calling this wrong.
```
Let's not give IPCs immunity to an ethereal demon.

Aurorablade commented:
Progress:Added in the Spell Exsanguinate
Spell Quirk: Exanguinate and whisper only work properly when phased in. I am assuming this is due to the fact when you phase out you transfer to a holder dummy. Will need to think on this when my brain is more active.(Read...Need Caffine...)

Removed IPC immunity and put in alot of robot gibs...NOMNOMNOM CRUNCHY CIRCUITS

Aurorablade commented:
DISREGARD SPELLS! USE VERBS! *CACKLES MADLY*

Fox-McCloud commented:
I...what.

Iamgoofball commented:
# RIP AND TEAR

## RIP AND TEAR YOUR GUTS

Aurorablade commented:
@Iamgoofball well the orignal intent from TG was admin spawn only (or suicidal wizard/Wizards giving item to apprentice/Admins spawning blood vial on unsuspecting Greytide) according to @Fox-McCloud  "No point to ading this if it's an admin only event; admins with necessary permissions can trigger events manually."

I am open to suggestions for anything though.

Disclaimer: Pun Pun WAS harmed mutliple times in my next commit (And i LIKE pun pun)

KorPhaeron commented:
@Iamgoofball 
@Aurorablade 

I wanted to test it first to see if it was suitable for being a normal random antag like ninja or aliens, so I set the event chance to zero for now

Aurorablade commented:
@KorPhaeron I hope you don't mind me running away with this adding things...

Aurorablade commented:
think i should borrow the fluff objective code from https://github.com/ParadiseSS13/Paradise/pull/1326 ? if so what would people like to see?

...And should this thing interact with the chaplain at all?

KorPhaeron commented:
Chaplain already interacts with cult, this thing interacts with janitors.

And no I don't mind you adding stuff, its the nature of open source.

Eating X number of people would be a good objective, or eating certain people.

Aurorablade commented:
latest commit:
Changed the GibEat() proc to use the mobs blood color. (defualts to human gibs if we eat Ian)
Added in (or tried to) so gib and innard throwing.
TRIED TO make sure we cannot sink into robot debris.

Reworked OBjectives slightly.

Should I consider not letting this thing EAT DEAD CYBORGS?!

Aurorablade commented:
so may add fluff objectives such as kill only this role, kill the janitor..or otherwise..Anyone else got fun suggestions for anything?

Aurorablade commented:
Ability Brainstorm, by suggestion: Devour so many people, Suddenly able to throw gibs at people while phased in/out...Just to make them bloody...and the shock of have gibs THROWN AT YOU

Aurorablade commented:
Added in a throwable gib move when you eat give people. 1 minute CD if i did it right. MAY need to tweak the throw_at speed...cuase at one point..I killed pun pun..with gibs....but then again i also had the CD off for testing...so i was spam throwing gibs...

Aurorablade commented:
Oh for the love of...BRB Making new branch
